File: src/examples/python/carrier_accounts/create_carrier_account.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_PROD_API_KEY')


carrier_account = easypost.CarrierAccount.create(
    type="UpsAccount",
    description="NY Location UPS Account",
    reference="my-reference",
    credentials={
        "account_number": "A1A1A1",
        "user_id": "USERID",
        "password": "PASSWORD",
        "access_license_number": "ALN"
    }
)

print(carrier_account)

File: src/examples/csharp/carrierAccounts/createCarrierAccount.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace csharp
{
    class createCarrierAccount
    {
        static async Task Main()
        {
            EasyPost.ClientManager.SetCurrent(Environment.GetEnvironmentVariable("EASYPOST_PROD_API_KEY"));

            Dictionary<string, object> credentials = new Dictionary<string, object>() {
                { "account_number", "A1A1A1" },
                { "user_id", "USERID" },
                { "password", "PASSWORD" },
                { "access_license_number", "ALN" }
            };

            CarrierAccount account = await CarrierAccount.Create(new Dictionary<string, object>() {
                { "type", "UpsAccount" },
                { "description", "description" },
                { "credentials", credentials }
            });

            Console.WriteLine(JsonConvert.SerializeObject(account, Formatting.Indented));
        }
    }
}
